app/src/org/commcare/adapters/NavDrawerAdapter.java

```diff
@@ -0,0 +1,43 @@
+package org.commcare.adapters;
+
+import android.content.Context;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import org.commcare.activities.components.NavDrawerItem;
+import org.commcare.dalvik.R;
+
+/**
+ * @author amstone326
+ */
+public class NavDrawerAdapter extends ArrayAdapter<NavDrawerItem> {
+
+    Context context;
+
+    public NavDrawerAdapter(Context context, NavDrawerItem[] drawerItems) {
+        super(context, android.R.layout.simple_list_item_1, drawerItems);
+        this.context = context;
+    }
+
+    @Override
+    public View getView(int position, View convertView, ViewGroup parent) {
+        View v = convertView;
+        if (v == null) {
+            v = View.inflate(context, R.layout.nav_drawer_item_view, null);
+        }
+        NavDrawerItem item = this.getItem(position);
```

Comment:
`getItem` has a `@Nullable` annotation, so you should probably check if it is null before using it